### ISSUE TITLE
Large set of various annotations for human  RPS27A  & UBA52  (Ubiquitin-ribosomal proteins)

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-annotation#1512]
Reactome

http://amigo.geneontology.org/amigo/gene_product/UniProtKB:P62979
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:P62987

About 90 annotations for both spanning many processes...many appear to be invalid?

v

**Comment by ValWood** [comment]
UBA52 also has ~300 annotations to nucleoplasm.

 This is a ribosomal protein fused to a ubiquitin (in genome). The ubiquitin is identical to "canonical" ubiquitin, so it could be that this protein is mistaken for ubiquitin, and then gets associated with many processes?

**Comment by deustp01** [comment]
No. In Reactome, we have created a protein "UBA52(1-76) (http://www.reactome.org/PathwayBrowser/#/R-HSA-69620&SEL=R-HSA-939203&PATH=R-HSA-1640170) corresponding to the first 76 residues of UniProt P62987 (http://www.uniprot.org/uniprot/P62987). P62987 is a polyprotein that can be cleaved to release a molecule of Ubiquitin (what we call UBA52(1-76)) and a molecule of 60S ribosomal protein L40. Our annotations pertain only to the former and its roles in ubiquitination.

Now, many of those annotations may in fact involve its role as one of the substrates in a ubiquitination process so we shouldn't be sending those annotations to GO, but that's a separate problem.

This is actually a big general problem for human ubiquitin - the identical 76-residue coding sequence occurs in several different polyproteins, each of which is presumed to be cleaved to release one or more functional ubiquitin protein molecules. We always identify and distinguish these by using the UniProt gene name plus the coordinates of the Ubiquitin protein within it. That works for us. Ideas about how to annotate that whole-protein versus protein-part distinction?

**Comment by ValWood** [comment]
Hi Peter,
I don't think that  P62987 is the ubiquitin poly protein.
If you follow the link through to Pfam,  in the UniProt entry domain section its the 
http://pfam.xfam.org/protein/P62987 ubiquitin fused to ribosomal L40e...
This is conserved from yeast to human. Pombe has one too:

Although until now I always thought that this ubiquitin was always  con-joined to the ribosomal protein. But SGD says that this one is also post-translationally cleaved. But in this case it will be a peptide bond, not a lysine...

![ubiquitin](https://cloud.githubusercontent.com/assets/7359272/22042746/619f5aee-dd04-11e6-88b2-f0645867bd33.jpg)

**Comment by ValWood** [comment]
I see what you mean....
We have the same problem with a number of other tandem fusions.
My  plan is to make the cleaved proteins the annotated objects, but we can't do this yet for most as we have no idea where they are cleaved.

This one is more complicated, it seems that the ubiquitin has some sort of chaperone role for ribosome assembly, then presumably after cleavage the Ub part joins the pool of ubiquitin?
I think I would only annotate  terms which apply to the fusion protein to P62979 though...
Having everything which applies to ubiquitin monomer is confusing.

**Comment by ValWood** [comment]
Deleted previous comment. I realise now that is exactly what you said.

So the issues here: 
i)  Short term, how to prevent every ubiquitin annotation becoming attached to the ribosomal-ubiquitin fusion (this fusion also has a function pre-cleavage in ribosome assembly).
ii) Longer term, how to represent these cleaved proteins.  In similar  cases the post-cleavage proteins can be represented individually, but the situation  here is more complicated because the pre-cleavage form has a function. Using PRO IDs for the post-cleavage forms would still propagate annotations (maybe there is a way to specify the relationship so that this does not occur).

**Comment by ValWood** [comment]
@deustp01  ant news here?

**Comment by pgaudet** [comment]
@deustp01 ping

**Comment by deustp01** [comment]
Val's two issues
i) preventing ubiquitin annotation from becoming attached to the ribosomal-ubiquitin fusion. As discussed above, we can do this within Reactome by annotating separately to the two cleavage products: ubiquitin annotations go to the ubquitin product and ribosome annotations go to the ribosome part. That still leaves two major problems. First, if the cleavage site is not known (and even in human I'm pretty sure it's inferred from staring at the sequence, not from studies of the enzymatic cleavage of purified fusion protein), as in fission yeast, this annotation-to-parts tactic won't work. Second, the community reference object that our two sets of annotations are applied to is the single UniProt record for the fusion protein so at the level of Reactome-to-UniProt mappings all our local distinctions get lumped back together again. Se have not tried to work with UniProt to sort that one pout. Suggestions on how to proceed are welcome.
ii) Nothing to add beyond noting overlap with i.

**Comment by pgaudet** [comment]
Thanks for the update @deustp01

**Comment by ValWood** [comment]
Is there a way to prevent any annotation transfer to the fusion proteins so that they were only annotated manually? (i.e.  have a list that the TAS etc did not apply to?). This could happen before export, or as part of the GO import pathway (some sort of annotation rule that this set of proteins  should only be annotated manually). 

We would need a good way to identify "this set"  but it could probably be done by PANTHER domain combination.

**Comment by pgaudet** [comment]
Interesting idea. @thomaspd Would Panther work for that ? 

Are there any kind of 'fusion' tag that we could use to create a check?

**Comment by ValWood** [comment]
Actually there is a bigger issue from the reactome data for things like ubiquitin

Check out the ribosome fusion protein. It has >500 annotations,  even once the incorrect ones are filtered there are many duplicates>

See
![nucleoplasm](https://user-images.githubusercontent.com/7359272/53655499-4b4a8e80-3c48-11e9-9a89-64d9875ca1f2.jpg)

there is a mapping from every pathway containing ubiquitin.
There are  around 100 annotations to "nucleoplasm". This isn't so useful for GO users. It seems that these are really "mappings to Reactome families" rather than annotations.
Wouldn't it be better for end users  if Reactome provided a non-redundant annotation set?


![nucleoplasm](https://user-images.githubusercontent.com/7359272/53655627-a11f3680-3c48-11e9-9930-6195d62a8a43.jpg)

and maybe block annotations from non-experimental sources for pthr10666?

**Comment by ValWood** [comment]
http://amigo.geneontology.org/amigo/gene_product/UniProtKB:P62979 has 327 annotations to cytosol!

**Comment by deustp01** [comment]
I'm confused. I just looked at all the DNA repair reactions at the top of your list. In every case a complex involved in a reaction includes ubiquitin as a component. Ubiquitin is a single polypeptide product that in humans appears to be encoded as a part of each of several polyproteins. UniProt determined several years ago, that if several genes encode the identical polypeptide, there shall be separate UniProt instances for each gene product. In the case of ubiquitin, I know of zero evidence to say which of these genes actually encodes the ubiquitin molecules used in these DNA repair related complexes. We have assembled a set, a list of all the possible souces of ubiquitin separated by logical ORs and annotated that set as the contributor of the ubiquitin found in the DNA repair related complexes.

So what are are doing wrong?

**Comment by ValWood** [comment]
The problem is that everything associated with "ubiquitin"  becomes associated with  the ribosomal-ubiquitin fusion (not only to the ubiquitin) and it isn't appropriate to assign every process ubiquitin is involved  in also to the fusion product which includes the ribosomal protein. 

but I'm not sure how this happens if  Reactome aren't explicitly exporting these annotations for P62979...unless the UniProt pipeline does this because this enty is 100% match to ubiquitin  (albiet only to a substring of this entire protein)

**Comment by deustp01** [comment]
We do export annotations to the fusion because the rules of the GAF game as we understand them say to associate the function of a covalently modified protein with the UniProt ID of its canonical form. If you need a different result, we need a different rule and that is the domain of GO and UniProt, not Reactome unless I'm missing something here. @pgaudet ?

**Comment by ukemi** [comment]
This is an interesting point in which the covalent modification is the ubiquitination. How do we treat the ubiquitin in these cases. How do we treat it in conventional GO annotations?

**Comment by ukemi** [comment]
This also came up peripherally when looking at the Reactome imports into GO-CAM. When a protein complex contains a ubquitinated protein, what is the role of the ubiquitin? Does it contribute to the complex just like all the other proteins?

**Comment by ukemi** [comment]
@deustp01 maybe the ribosomal fusion form shouldn't be included in the Reactome ubiquitin sets if it isn't used equivalently to the other ubiquitins?

**Comment by deustp01** [comment]
But it is used equivalently. From the [UniProt entry](https://www.uniprot.org/uniprot/P62987) for human Ubiquitin-60S ribosomal protein L40:

> Ubiquitin is encoded by 4 different genes. UBA52 and RPS27A genes code for a single copy of ubiquitin fused to the ribosomal proteins L40 and S27a, respectively. UBB and UBC genes code for a polyubiquitin precursor with exact head to tail repeats, the number of repeats differ between species and strains.

So in **_all_** cases for humans and probably at least all vertebrates, as far as anyone knows, a germline gene encodes a polypeptide that is post-translationally cleaved to yield one or more ubiquitin polypeptides and possibly a polypeptide with a different sequence and function. Within Reactome, when annotating ubiquitin functions, we specify the UniProt entry for the full-length protein coding gene and use our own modification attributes to indicate that we are referring to the contiguous subset of residues that specifies ubiquitin (or the subset that specifies a ribosomal protein if we're annotating ribosomal protein functions).

The problem, again. is that when that specific, correct annotation is exported, the annotation that links the specific function to that specific subregion of the protein is lost. Figure out how to preserve that information and the problem goes away. Insist that the unit of function:protein mapping is GO is the canonical UniProt full-length sequence and the problem persists.

**Comment by pgaudet** [comment]
This came up before also when annotating viral proteins (since they are encoded by a single transcript): perhaps annotating to UniProt 'chains' would solve the problem? 
<img width="1014" alt="image" src="https://user-images.githubusercontent.com/4782928/53741145-fd2aca80-3e95-11e9-91c5-434158bb606c.png">

**Comment by deustp01** [comment]
Interesting idea. Immediate problem: the fact that a full-length translation product is always cleaved to generate mature polypeptides does not always lead to a "chains" annotation in the UniProt record for the full-length entity. E.g., [insulin](https://www.uniprot.org/uniprot/P01308) (perhaps that omission is reasonable) but also [POMC (pro-opiomelanpcortin)](https://www.uniprot.org/uniprot/P01189#ptm_processing), the precursor of ACTH and other polypeptide hormones generated by POMC cleavage. In fact, it's worse than not captured. It is captured but as a "peptide" attribute under "molecular processing" rather than as a "chains" attribute.

However, to patch specific annotations when a "chains" or "peptides" attribute exists, is there documentation someplace to explain how to incorporate this information into a GAF or GPAD entry that now points to a protein like UBA52 to redirect the  pointer specifically to its ubiquitin part?

**Comment by pgaudet** [comment]
I thought it could be handled the same way we handle isoforms.

**Comment by deustp01** [comment]
Hmmm. Isoforms get different UniProt IDs: P12345 has isoforms P12345-1, P12345-2, and so on. I don't see how to generate the IDs to point to the ubiquitin and ribosomal protein parts of UBA52, but I may be missing something here.

**Comment by pgaudet** [comment]
The chains also have IDs: 
Ubiquitin: PRO_0000396433 
Ribosomal protein L40: PRO_0000396434
(see screenshot above)

**Comment by deustp01** [comment]
And how do those IDs relate back to UniProt P62987? Or can we rely on users not to want  to know about this relationship?

**Comment by pgaudet** [comment]
I thought IDs were captured in Col1 and isoforms in Col17 - this is how I thought it should relate. That being said, at the display level, it's hard to make the information clear.

**Comment by ValWood** [comment]
> We do export annotations to the fusion because the rules of the GAF game as we understand them say to associate the function of a covalently modified protein with the UniProt ID of its canonical form.

@deustp01  you are correct, and I realised as this thread trundled on that the way we are supposed to annotate tandem fusions  would result  in the annotations  applicable to the individual cleaved proteins would be assigned to both parts. The ribosome-ubiquitin fusion is a 'special' case because this is picking up entries assigned to the individual ubiquitin molecule and  popping them back onto the full length ribosome-ubiquitin fusion.

In other  tandem fustion cases, PomBase don't represent the individual 
but  have both processess assigned to the full length product-  until we  know the cleavage site we can't do a better job and  annotatate the individual gene products.

For the ribosome-ubiquitin fusion, I worked around this by only assigning the roles of the full-length protein using GO, (because the ubiquitin  isn't immediately cleaved but has a role in  facilitating the assembly of the ribosome).

The ubiquitin hybrid gene UBA52 regulates ubiquitination of ribosome and sustains embryonic development.

Abstract
Ubiquitination is a crucial post-translational modification; however, the functions of ubiquitin-coding genes remain unclear. UBA52 encodes a fusion protein comprising ubiquitin at the N-terminus and ribosomal protein L40 (RPL40) at the C-terminus. Here we showed that Uba52-deficient mice die during embryogenesis. UBA52-deficient cells exhibited normal levels of total ubiquitin. However, UBA52-deficient cells displayed decreased protein synthesis and cell-cycle arrest. The overexpression of UBA52 ameliorated the cell-cycle arrest caused by UBA52 deficiency. Surprisingly, RPL40 expression itself is insufficient to regulate cyclin D expression. The cleavage of RPL40 from UBA52 was required for maintaining protein synthesis. Furthermore, we found that RPL40 formed a ribosomal complex with ubiquitin cleaved from UBA52. UBA52 supplies RPL40 and ubiquitin simultaneously to the ribosome. Our study demonstrated that the ubiquitin-coding gene UBA52 is not just an ubiquitin supplier to the ubiquitin pool but is also a regulator of the ribosomal protein complex. These findings provide novel insights into the regulation of ubiquitin-dependent translation and embryonic development.

So although it does contribute to the ubiquitin pool, until we can  annotate the ubiquitin in this hybrid protein independently, it made sense to me not to apply these GO terms to the fusion protein (because we have ubiquitin from the major polyprotein which holds all of this annotation).

This is more of a work-around than a robust solution though ;)
